worker/lease/manager.go

```diff
@@ -292,14 +292,6 @@ func (manager *Manager) retryingClaim(claim claim) {
 			claim.respond(lease.ErrClaimDenied)
 			return
 		}
-		// now, this isn't strictly true, as the lease can be given for longer
-		// than the requested duration. However, it cannot be shorter.
-		// Doing it this way, we'll wake up, and then see we can sleep
-		// for a bit longer. But we'll always wake up in time.
-		manager.ensureNextTimeout(claim.duration)
```

Comment:
We don't need to do this, because we are always guaranteed to have a check scheduled sooner, at which point we will check, then calculate an appropriate next tick.